File: refactor_router.js
const fs = require('fs');

let content = fs.readFileSync('src/router/index.js', 'utf8');

// Regex to find all static component imports (except layouts or special ones if needed, but we only have pages and auth)
// Examples:
// import Home from '../pages/Home.vue'
// import Product from '../pages/Product.vue'
// import EducationTraining from '../pages/services/EducationTraining.vue'

const importRegex = /import\s+([A-Za-z0-9_]+)\s+from\s+['"]\.\.\/pages\/([^'"]+\.vue)['"]/g;

let match;
const routesMap = {};

// Find them all and populate the map
while ((match = importRegex.exec(content)) !== null) {
    const componentName = match[1];
    const componentPath = match[2];
    routesMap[componentName] = `../pages/${componentPath}`;
}

// Remove the static imports
content = content.replace(/import\s+[A-Za-z0-9_]+\s+from\s+['"]\.\.\/pages\/[^'"]+\.vue['"]\s*\n/g, '');

// Replace the component: ComponentName with component: () => import(...)
for (const [componentName, componentPath] of Object.entries(routesMap)) {
    // We only want to match exactly `component: ComponentName,` or `component: ComponentName\n`
    // Not `component: () => import(...)` which already exists for some
    const compRegex = new RegExp(`component:\\s*${componentName}(,|\\s)`, 'g');
    content = content.replace(compRegex, `component: () => import('${componentPath}')$1`);
}

fs.writeFileSync('src/router/index.js', content);
console.log("Router refactored to use dynamic imports");

File: refactor_router.py
import re

with open('src/router/index.js', 'r', encoding='utf-8') as f:
    content = f.read()

# Regex to find all static component imports
# e.g., import Home from '../pages/Home.vue'
import_regex = re.compile(r"import\s+([A-Za-z0-9_]+)\s+from\s+['\"](\.\./pages/[^'\"]+\.vue)['\"]")

routes_map = {}
for match in import_regex.finditer(content):
    component_name = match.group(1)
    component_path = match.group(2)
    routes_map[component_name] = component_path

# Remove the static imports
content = re.sub(r"import\s+[A-Za-z0-9_]+\s+from\s+['\"]\.\./pages/[^'\"]+\.vue['\"]\s*\n", '', content)

# Replace the component: ComponentName with component: () => import(...)
for comp_name, comp_path in routes_map.items():
    comp_regex = re.compile(r"component:\s*" + comp_name + r"(,|\s)")
    content = comp_regex.sub(rf"component: () => import('{comp_path}')\g<1>", content)

with open('src/router/index.js', 'w', encoding='utf-8') as f:
    f.write(content)

print("Router refactored to use dynamic imports")
